src/main/java/dukeApp/files/Task.java

```diff
@@ -0,0 +1,59 @@
+package dukeApp.files;
+
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task() {
+        //empty
+    }
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns a tick if task is done and cross if otherwise
+     * @return a tick or cross icon
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); //return tick or X symbols
+    }
+
+    /**
+     * Mark task as done
+     */
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    public abstract String getType();
+
+    public abstract String getTask();
+
+    public abstract String getDate();
```

Comment:
It would be better not to provide getDate, getTime etc. methods if they're not needed in every subclass. E.g. Todo does not need these methods but will have to implement them because it is stated here, which does not make sense. Instead, add the specific methods in the subclasses that require them.